miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.roboconf.doc.generator.internal.renderers;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.List;
+
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.xml.sax.SAXException;
+
+import net.roboconf.core.model.beans.ApplicationTemplate;
+import net.roboconf.core.utils.Utils;
+import net.roboconf.doc.generator.RenderingManager;
+import net.roboconf.doc.generator.RenderingManager.Renderer;
+import net.roboconf.doc.generator.internal.AbstractStructuredRenderer;
+
+/**
+ * A renderer that outputs pdf.
+ * @author Amadou Diarra - UGA
+ */
+public class PdfRenderer extends AbstractStructuredRenderer {
+	
+	/**
+	 * Constructor.
+	 * @param outputDirectory
+	 * @param applicationTemplate
+	 * @param applicationDirectory
+	 */
+	public PdfRenderer(File outputDirectory, ApplicationTemplate applicationTemplate, File applicationDirectory) {
+		super(outputDirectory, applicationTemplate, applicationDirectory);
+	}
+
+	@Override
+	protected File writeFileContent(String fileContent) throws IOException {
+		
+		//Generate fop render
+		new RenderingManager().render( this.outputDirectory, this.applicationTemplate, this.applicationDirectory, Renderer.FOP, options );
+		File index_fo = new File( this.outputDirectory, "index.fo" );
+		File index_pdf = new File( this.outputDirectory, "index.pdf" );
+		
+		//Copy fop configuration file in outputDirectory
+		InputStream conf = getClass().getResourceAsStream( "/fop.xconf" );
+		File fopConfig = new File( this.outputDirectory, "fop.xconf" );
+		Utils.copyStream( conf, fopConfig );
+		
+		//Generate pdf render
+		OutputStream out = null;
+		try {
+				out = new BufferedOutputStream( new FileOutputStream(index_pdf) );
+				FopFactory fopFactory = FopFactory.newInstance( fopConfig );
+				Fop fop =  fopFactory.newFop( "application/pdf", out );
+				Source src = new StreamSource( index_fo );
+
+				TransformerFactory factory = TransformerFactory.newInstance();
+				Transformer transformer = factory.newTransformer();
+
+				Result res = new SAXResult(fop.getDefaultHandler());
+				transformer.transform(src, res);
+		} catch (SAXException e) {
```

Comment:
Wrap all your catch blocks within a single one.
And propagate them as an IO exception.

```java
try {

} catch( Exception e) {
         throw new IOException( e );

} finally {

}
```